orangecontrib/bio/resolwe/genapi.py

```diff
@@ -0,0 +1,110 @@
+"""Genapi"""
+import json
+import io
+import gzip
+import requests
+
+from genesis import Genesis, GenData
+from Orange.data import ContinuousVariable, StringVariable, Domain, Table
+from .. import dicty
+
+DEFAULT_EMAIL = 'anonymous@genialis.com'
+DEFAULT_PASSWD = 'anonymous'
+DEFAULT_URL = 'https://dictyexpress.research.bcm.edu'
+CallBack = dicty.CallBack
+
+
+class Genapi(object):
+    """
+    Python module that leverages Genesis PyAPI (Python API for accsess to DictyExpress database).
+    It supports connection to the server and data retrieval functionalities.
+    """
+    def __init__(self, email=DEFAULT_EMAIL, password=DEFAULT_PASSWD, url=DEFAULT_URL):
+
+        self._gen = Genesis(email, password, url)
+        self.email = email
+
+    def fetch_etc_objects(self, callback=lambda: None):
+        """
+        Function downloads all available :obj:`GenData` etc objects from DictyExpress database.
+
+        :return: list of :obj:`GenData` objects.
+
+        """
+        cbc = CallBack(1, callback, callbacks=100)
+        try:
+            list_of_experiments = self._gen.api.data.get(case_ids__contains='5535115cfad58d5e03006217', status='done',
+                                                         type__startswith='data:etc:')['objects']
+
+        except requests.exceptions.ConnectionError:
+            raise Exception('Server not accessible, check your connection.')
+
+        else:
+            cbc()
+            store_experiments = [GenData(exp, self._gen) for exp in list_of_experiments]
+            cbc.end()
+            return store_experiments
+
+    def download_etc_data(self, gen_data_id, callback=lambda: None):
+        """
+        Function downloads etc data of a chosen experiment from the server.
+
+        :param gen_data_id: id of :obj:`GenData` object to download.
+        :return: Returns data in json like format
+        :rtype: :obj:`dict`
+
+        """
+        cbc = CallBack(3, callback, callbacks=50)
+
+        try:
+            response = next(self._gen.download([gen_data_id], 'output.etcfile'))
+        except requests.exceptions.ConnectionError:
+            raise Exception('Server not accessible, check your connection.')
```

Comment:
Create a new type of exception, e.g.
```py
class GenAPIException(Exception):
    pass
```
or a class of exceptions
```py
class GenAPIHTTPException(GenAPIException):
    pass

class GenAPISomeOtherException(GenAPIException):
    pass
```
And raise that/those.